### PR TITLE
Zy/563 forest add configuration

### LINKED ISSUE #563: FOREST: Add configuration
Image

### PR DESCRIPTION
**Pull request by zheyxu**
# Pull Request Details

## What issue does this PR address

- FOREST: Add configuration

- Link to associated issue: #563 

## Checklist

Complete the check-list below to ensure your branch is ready for PR.

Flutter Style Guide: https://survivor.togaware.com/gnulinux/flutter-style.html

- [x] Screenshots included in linked issue
- [x] Changes adhere to the style and coding guideline
- [x] No confidential information
- [x] No duplicated content
- [x] No lint check errors related to your changes (`make prep` or `flutter analyze lib`)
- [x] Pre-exisiting lint errors noted: [HERE]
- [x] Integration test `make qtest.tmp` screenshot included in issue
- [x] Tested on device:
  - [ ] Linux
  - [x] MacOS
  - [ ] Windows
- [x] Added two reviewers

## Finalising

Once PR discussion is complete and reviewers have approved:

- [ ] Merge dev into the branch
- [ ] Resolve any conflicts
- [ ] Add one line summary into CHANGELOG.md
- [ ] Bump appropriate version number in pubspec.yaml
- [ ] Push to git repository and review
- [ ] Merge PR into dev (gjwgit)

### CODE REVIEW

File: assets/r/model_build_conditional_forest.R [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+# Author: Graham Williams
+
+# Random Forest using randomForest()
```
Comment (gjwgit) [inline assets/r/model_build_conditional_forest.R]: Isn't this now using conditional forest in this script?

File: assets/r/model_build_conditional_forest.R [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+response_variable <- all.vars(form)[1]
+
+# Remove observations with missing values in the response variable.
+
+tds_clean <- tds[!is.na(tds[[response_variable]]), ]
+
+# Build the conditional random forest model using tds_clean.
+
+model_conditionalForest <- cforest(
+  form,
```
Comment (gjwgit) [inline assets/r/model_build_conditional_forest.R:47]: As standard practise please align = here/

File: assets/r/model_build_conditional_forest.R [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+library(rattle)
+library(party)
+library(reshape2)
+
+mtype <- "conditionalForest"
+mdesc <- "Forest"
+
+# Extract the response variable name from the formula.
+
+response_variable <- all.vars(form)[1]
```
Comment (gjwgit) [inline assets/r/model_build_conditional_forest.R]: In other scripts don't we use `target` to get the response variable?

**Review by gjwgit — COMMENTED** [review]
This doesn't look right?

image

File: lib/features/forest/config.dart [hunk 4]
```diff
@@ -25,14 +25,31 @@
...
+import 'package:rattle/widgets/labelled_checkbox.dart';
+import 'package:rattle/widgets/number_field.dart';
+import 'package:rattle/widgets/vector_number_field.dart';
+
+/// Descriptive tooltips for different algorithm types,
+/// explaining the splitting method and potential biases.
+
+Map forestTooltips = {
+  AlgorithmType.conditional:
+      'Build multiple decision trees using random samples of data and features, then aggregate their predictions.',
```
Comment (gjwgit) [inline lib/features/forest/config.dart]: These lines are too long - please fix and don't repeat.

**Review by gjwgit — COMMENTED** [review]
Please fix all the SizedBox() to use the standard constants as you come across them please.

File: lib/features/forest/config.dart [hunk 5]
```diff
@@ -65,17 +101,115 @@ class ForestConfigState extends ConsumerState<ForestConfig> {
...
+                      1,
+                      duration: const Duration(milliseconds: 300),
+                      curve: Curves.easeInOut,
+                    );
               },
               child: const Text('Build Random Forest'),
             ),
             configWidgetGap,
             Text('Target: ${getTarget(ref)}'),
+            configWidgetGap,
```
Comment (gjwgit) [inline lib/features/forest/config.dart:140]: Please follow the same style of layout where possible - so a blank line around the gap - this is consistent everywhere please.

**Comment by zheyxu** [comment]
> This doesn't look right?
> 
> image

I think it looks correct. May need further discussion.

**Pull request merged**